### ISSUE TITLE
Fuzz testing

### ISSUE THREAD

**Issue by ericdill** [NSLS-II/bluesky#297]
Quoted from @danielballan in #289
>
>On Slack, I suggested that it would be interesting to "noise test" the RunEngine: that is, generate random sequences of Messages and make sure the RunEngine always lands in a recoverable state. Maybe py.test offers nice ways to do this kind of thing.
>
>To be clear, I am not suggesting that this belongs in our automated test suite. We do not want randomized tests running on Travis. Rather it should be a script that we run manually -- perhaps 100, 1000, 10000 times -- to discover edge cases, leading to bug fixes and then *targeted* tests for those cases in the automated test suite.
>
>Something like:
>
>```python
def smoke_test(plan)
    try:
        RE = setup_test_run_engine()
        RE(plan)
        while True:
            if RE.state == 'paused':
                RE.resume()
            elif RE.state == 'idle':
                break
    except IllegalMessageSequnece:
        # This is the only kind of error we should except. Swallow it.
        # Any other errors probably indicate a flaw in the RunEngine's logic.
>
for i in range(100):
    random_plan = ...
    try:
        noise_test(plan)
    except Exception:
        # re-raise for inspection, or log the traceback and `random_plan` to a file
```

**Comment by ericdill** [comment]
Notes to self for later...

All valid messages can be found here: RunEngine._command_registry

```python
from bluesky.tests import utils
RE = utils.setup_test_run_engine()
RE._command_registry
```
```
{'checkpoint': <bound method RunEngine._checkpoint of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'close_run': <bound method RunEngine._close_run of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'collect': <bound method RunEngine._collect of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'configure': <bound method RunEngine._configure of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'create': <bound method RunEngine._create of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'kickoff': <bound method RunEngine._kickoff of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'null': <bound method RunEngine._null of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'open_run': <bound method RunEngine._open_run of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'pause': <bound method RunEngine._pause of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'read': <bound method RunEngine._read of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'save': <bound method RunEngine._save of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'set': <bound method RunEngine._set of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'sleep': <bound method RunEngine._sleep of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'stage': <bound method RunEngine._stage of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'subscribe': <bound method RunEngine._subscribe of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'trigger': <bound method RunEngine._trigger of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'unstage': <bound method RunEngine._unstage of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'unsubscribe': <bound method RunEngine._unsubscribe of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'wait': <bound method RunEngine._wait of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>,
 'wait_for': <bound method RunEngine._wait_for of <bluesky.run_engine.RunEngine object at 0x7f1001f64fd0>>}
```

**Comment by danielballan** [comment]
Closed by #380
